### PR TITLE
Replace lookup button with SERP based definition

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/42792087274227/1200417326972258
Tech Design URL:
CC:

**Description**:

Replace lookup button with SERP based definition

**Steps to test this PR**:

1.   Open a page, select some text when the menu appears choose lookup.

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [x] iPhone X
* [x] iPad

**OS Testing**:

* [x] iOS 13
* [x] iOS 14

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

File: DuckDuckGo/DefineTermViewController.swift [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        webView.navigationDelegate = self
+        view.addSubview(webView)
+
+        if let term = term {
+            load(term: term)
+        }
+    }
+
+    func load(term: String) {
+        print("***", #function, term)
```
Comment (samsymons) [inline DuckDuckGo/DefineTermViewController.swift]: Debugging call leftover here 🙂

File: Core/WebView.swift [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+}
+
+public class WebView: WKWebView {
+
+    public weak var webViewDelegate: WebViewDelegate?
+
+    // swiftlint:disable identifier_name
+    @IBAction func _lookup(_ sender: UIMenuController?) {
+        evaluateJavaScript("window.getSelection().toString()") { result, _ in
```
Comment (samsymons) [inline Core/WebView.swift]: A potential issue with this API is that it returns an empty string if you call it on any element within an iframe ([MDN](https://developer.mozilla.org/en-US/docs/Web/API/Window/getSelection)):

>When called on an <iframe> that is not displayed (eg. where display: none is set) Firefox will return null, whereas other browsers will return a Selection object with Selection.type set to None.

I took a quick go at checking whether `term` is empty and calling the superclass's implementation of this if it is, but it didn't show the typical lookup view.

**Review by samsymons — APPROVED** [review]
LGTM! Two comments here, not certain how much of an issue the `iframe` thing is.

File: Core/WebView.swift (hunk 2 above)
Comment (brindy) [reply Core/WebView.swift]: If the frame is not displayed how would the user select text and then show the popover menu? 🤔 

I think this should be fine, but nice attention to detail.

Comment (samsymons) [reply Core/WebView.swift]: The case I tested with is [here](https://www.w3schools.com/html/html_iframe.asp), if you use Lookup on a string within that nested page it'll come back with an empty string. But again I really don't know how much of a problem this is. 😄

Comment (brindy) [reply Core/WebView.swift]: I see now - yeah I need to either do nothing if the text is empty or have another method for getting the selection :/

Comment (brindy) [reply Core/WebView.swift]: I've added another user script to detect selections.

@SlayterDev would you mind giving this a quick review as well, please?

**Comment by brindy** [comment]
@samsymons I ended up doing some minor refactoring around user scripts so I could re-use the logic.  Can you give that a bit of test when you come to re-review this, please? (in particular check I'm not leaking memory)

File: Core/WebView.swift (hunk 2 above)
Comment (samsymons) [reply Core/WebView.swift]: This looks good to me! :+1:

**Review by SlayterDev — APPROVED** [review]
JS looks good 👍

**Comment by brindy** [comment]
We decided this isn't a great change and could have wider implications.  Thanks for the reviews though.

**Pull request closed without merging**